pkg/daemon/ceph/osd/volume.go

```diff
@@ -149,12 +153,28 @@ func (a *OsdAgent) initializeDevices(context *clusterd.Context, devices *DeviceO
 		// Reporting
 		reportArgs := append(batchArgs, []string{
 			"--report",
+			"--format",
+			"json",
 		}...)
 
-		if err := context.Executor.ExecuteCommand(false, "", baseCommand, reportArgs...); err != nil {
+		cvOut, err := context.Executor.ExecuteCommandWithOutput(false, "", baseCommand, reportArgs...)
+		if err != nil {
 			return fmt.Errorf("failed ceph-volume report. %+v", err) // fail return here as validation provided by ceph-volume
 		}
 
+		logger.Infof("ceph-volume report: %+v", cvOut)
+
+		var cvReport cephVolReport
+		if err = json.Unmarshal([]byte(cvOut), &cvReport); err != nil {
+			return fmt.Errorf("failed to unmarshal ceph-volume report json. %+v", err)
+		}
+
+		for _, osd := range cvReport.Osds {
+			if len(osd.BlockDB) == 0 {
+				return fmt.Errorf("ceph-volume not using %s as metadataDevice.", a.metadataDevice)
```

Comment:
Error messages shouldn't end in a period.

I think this message needs to be more verbose for users to appropriately understand and correct the issue. Something like: "ceph-volume did not use the expected metadataDevice [%s]. ceph-volume report: %s". Ideally, we should also output why c-v didn't do it correctly, but I'm not sure if that information is reported.